### PR TITLE
Remove traces from jquery-detached from core

### PR DESCRIPTION
**Pull request by fqueiruga**
Follow up cleanup to https://github.com/jenkinsci/jenkins/pull/4929

This PR removes some dependencies that cause a _WEB-INF/lib/jquery-detached-1.2.1-core-assets.jar_ to be included on the a compiled Jenkins _war_. 

No issues when manually testing so far.

### Proposed changelog entries

* Remove unused jquery dependencies.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@timja 
@oleg-nenashev 
@Wadeck 

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by timja — APPROVED** [review]
💣

**Review by res0nance — APPROVED** [review]

**Review by Wadeck — APPROVED** [review]
Not manually tested, please go forward 👍

**Comment by timja** [comment]
AssetManagerTest needs reworking as it was using jquery to test that it worked

**Comment by fqueiruga** [comment]
TBH I don't see any way to make this test make sense. I'm replacing it with the handlebars one, but that can also go away sometime.

**Comment by timja** [comment]
> TBH I don't see any way to make this test make sense. I'm replacing it with the handlebars one, but that can also go away sometime.

can you add a test asset for it?

maybe in something like `src/test/webapp`?

**Comment by fqueiruga** [comment]
I think the AssetManager class does not deal with static assets, it's just to fish out stuff within plugins. https://github.com/jenkinsci/jenkins/blob/f7884d3513c6e436baef432a486197a3bb4d846c/core/src/main/java/jenkins/model/AssetManager.java#L23. Maybe I misunderstand it though. I'll be replacing the test to look for handlebars

File: test/src/test/java/jenkins/model/AssetManagerTest.java
Comment (timja) [inline test/src/test/java/jenkins/model/AssetManagerTest.java:52]:
```suggestion
```

**Comment by fqueiruga** [comment]
Thank you for the suggestion. If this build passes I'll take it out of draft.

**Review by timja — APPROVED** [review]

**Comment by batmat** [comment]
FYI, this PR got put out of draft 3 hours ago. So if the status is actually reviewed and approved, like it seems, we could probably mark it as `ready-for-merge`?

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

File: war/pom.xml
Comment (jglick) [inline war/pom.xml:143]: It is intentional that this one “core asset” remains?

Comment (timja) [reply war/pom.xml:143]: it's used in setup wizard and in plugin manager

Comment (jglick) [reply war/pom.xml:143]: OK, just checking.

**Pull request merged**
